app/src/main/java/com/example/duccm/learningapp/MainActivity.java

```diff
@@ -23,16 +32,89 @@ protected void onCreate(Bundle savedInstanceState) {
         initViews();
 
         setSupportActionBar(toolbar);
+        getSupportActionBar().setTitle(R.string.timeline);
 
-        PagerAdapter adapter = new PagerAdapter(getSupportFragmentManager());
+        adapter = new PagerAdapter(getSupportFragmentManager());
         viewPager.setAdapter(adapter);
 
         tabLayout.setupWithViewPager(viewPager);
+
+        viewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
+            @Override
+            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
+
+            }
+
+            @Override
+            public void onPageSelected(int position) {
+                invalidateOptionsMenu();
+            }
+
+            @Override
+            public void onPageScrollStateChanged(int state) {
+
+            }
+        });
     }
 
     public void initViews() {
         toolbar = (Toolbar) findViewById(R.id.toolbar);
         tabLayout = (TabLayout) findViewById(R.id.tab_layout);
         viewPager = (ViewPager) findViewById(R.id.pager);
     }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.menu_main, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onPrepareOptionsMenu(Menu menu) {
+        tabSelected = viewPager.getCurrentItem();
+        switch (tabSelected) {
+            case 0:
+                break;
+            case 1:
+                menu.clear();
+                break;
+        }
+        return super.onPrepareOptionsMenu(menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.add_post:
+                showDialog();
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    public void showDialog() {
+        PostStatusDialogFragment dialog = new PostStatusDialogFragment();
+        dialog.show(getSupportFragmentManager(), "dialog");
+
+    }
+
+    @Override
+    public void passData(Post post) {
+        FragmentManager fm = this.getSupportFragmentManager();
+        Fragment dialogFragment = fm.findFragmentByTag("dialog");
+        if (dialogFragment !=null) {
```

Comment:
chô này nên check thêm dialogFragment instance of PostStatusDialogFragment để tránh class cast exception